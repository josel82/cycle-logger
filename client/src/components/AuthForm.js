import React, { Component } from 'react';

class AuthForm extends Component {

    constructor(props){
        super(props);
        this.state = {
            email: '',
            password: '',
            emailInvalid: false,
            passwordInvalid: false
        }
    }
    onEmailChange = (email) => {
        this.setState({email});
        if(email.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i)){
            this.setState({emailInvalid:false});
        }else{
            this.setState({emailInvalid:true});
        }
    }
    onPasswordChange = (password) => {        
        this.setState({password});
        if(password.match(/^.{8,32}$/)){
            this.setState({passwordInvalid:false});
        }else{
            this.setState({passwordInvalid:true});
        }
    }
    handleOnSubmit = (e) => {
        e.preventDefault();
    
        this.props.handleAuthorization(this.state.email, this.state.password);
    }

    render(){
        return (
            <div className="form-container">
                <h1 className="form-title">{this.props.action}</h1>
                <div className="form-box">
                    <form onSubmit={this.handleOnSubmit} className="form form-auth">
                        <input 
                            type="text" 
                            className="form__input" 
                            name="email" 
                            value={this.state.email}
                            onChange={this.onEmailChange}
                             />
                        <input 
                            type="password" 
                            className="form__input" 
                            name="password" 
                            value={this.state.password}
                            onChange={this.onPasswordChange} 
                            />
                        <button 
                            className="btn btn-primary btn-md"
                            disabled={this.state.emailInvalid || this.state.passwordInvalid}
                            >
                            {this.props.action}
                        </button>
                    </form>
                </div>
            </div>  
        );
    }
}

export default AuthForm;