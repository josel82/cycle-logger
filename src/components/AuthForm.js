import React, { Component } from 'react';

class AuthForm extends Component {

    constructor(props){
        super(props);
        this.handleOnSubmit = this.handleOnSubmit.bind(this);
    }

    handleOnSubmit(e){
        e.preventDefault();

        const email = e.target.elements.email.value;
        const password = e.target.elements.password.value;
    
        this.props.handleAuthorization(email, password);
    }

    render(){
        return (
            <div class="form-container">
                <form onSubmit={this.handleOnSubmit} className="form-style-column">
                    <input type="text" className="form-input" name="email" id="email" />
                    <input type="password" className="form-input" name="password" id="password" />
                    <button className="btn-primary">{this.props.action}</button>
                </form>
            </div>  
        );
    }
}

export default AuthForm;