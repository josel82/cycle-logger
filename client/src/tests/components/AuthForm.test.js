import React from 'react';
import { shallow } from 'enzyme';

import AuthForm from '../../components/AuthForm';

let wrapper;

beforeEach(()=>{
    wrapper = shallow(<AuthForm />);
});

test('should render AuthForm component correctly', ()=>{
    expect(wrapper).toMatchSnapshot();
});

test('should handle onEmailChange', ()=>{
    const email = 'test@email.com';
    wrapper.find('input').at(0).prop('onChange')(email);
    expect(wrapper.state('email')).toBe(email);
});

test('should disable button if invalid email', ()=>{
    const invalid = 'test';
    wrapper.find('input').at(0).prop('onChange')(invalid);
    expect(wrapper.state('emailInvalid')).toBe(true);
    expect(wrapper.find('button').prop('disabled')).toBe(true);
});

test('should enable button if valid email', ()=>{
    const valid = 'test@email.com';
    wrapper.find('input').at(0).prop('onChange')(valid);
    expect(wrapper.state('emailInvalid')).toBe(false);
    expect(wrapper.find('button').prop('disabled')).toBe(false);
});

test('should handle onPasswordChange', ()=>{
    const password = 'Mypassword';
    wrapper.find('input').at(1).prop('onChange')(password);
    expect(wrapper.state('password')).toBe(password);
});

test('should disable button if invalid password', ()=>{
    const invalid = 'weak';
    wrapper.find('input').at(1).prop('onChange')(invalid);
    expect(wrapper.state('passwordInvalid')).toBe(true);
    expect(wrapper.find('button').prop('disabled')).toBe(true);
});

test('should enable button if valid password', ()=>{
    const valid = 'hdgs7262g2';
    wrapper.find('input').at(1).prop('onChange')(valid);
    expect(wrapper.state('passwordInvalid')).toBe(false);
    expect(wrapper.find('button').prop('disabled')).toBe(false);
});

describe('form submission', ()=>{
    test('should handle on submit', ()=>{
        const handleOnSubmit = jest.fn();
        const authData = {
            email: 'test@email.com',
            password: '1234abcd'
        }
        
        const wrapper = shallow(<AuthForm handleAuthorization={handleOnSubmit}/>);
        wrapper.find('input').at(0).prop('onChange')(authData.email);
        wrapper.find('input').at(1).prop('onChange')(authData.password);
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {}
        });
        expect(handleOnSubmit).toHaveBeenLastCalledWith(authData.email,authData.password);
    });
});