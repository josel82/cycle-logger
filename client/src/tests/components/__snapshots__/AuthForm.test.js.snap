// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AuthForm component correctly 1`] = `
<div
  className="form-container"
>
  <h1
    className="form-title"
  />
  <div
    className="form-box"
  >
    <form
      className="form form-auth"
      onSubmit={[Function]}
    >
      <input
        className="form__input"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        className="form__input"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <button
        className="btn btn-primary btn-md"
        disabled={false}
      />
    </form>
  </div>
</div>
`;
